pulsar-websocket/src/main/java/org/apache/pulsar/websocket/admin/WebSocketWebResource.java

```diff
@@ -81,9 +83,16 @@ public String clientAppId() {
         return clientId;
     }
 
-    public AuthenticationDataHttps authData() {
+    public AuthenticationDataSource authData() throws AuthenticationException {
```

Comment:
Instead of updating this method, it might make more sense to update the `clientAppId()` method in this class so that it sets the `authData`. (We'd need to rename it to indicate that it is authenticating the request, but I think that's a reasonable change, given that the method is already doing authentication.) That method is already completing authentication and calls `service().getAuthenticationService().authenticateHttpRequest(httpRequest)` currently. The current design will lead to double authentication.